### DISCUSSION TITLE
When using with a shared library I get undefined symbol!

### DISCUSSION THREAD

**Discussion by nonunknown** [DanielChappuis/reactphysics3d#317] [Q&A]
I'm trying to use this library with godot, for that I build a shared library and link with react physics's static library, but I get this error:

undefined symbol: _ZTIN14reactphysics3d14CollisionShapeE

I tried to find this symbol in the static library using: 

` nm -s ZTIN14reactphysics3d14CollisionShapeE libreactphysics3d.a `

and it is in the static library:
![image](https://user-images.githubusercontent.com/58845030/217606500-95a97631-1bc9-4b72-8ada-43eeb24e9584.png)

By reading in the internet, I found out maybe its a version mismatch between shared library and static library, my dinamic one is C++ 17 and -fPIC while seems the react phy is c++11

any tough what I can do?

**Comment by lyuma** [comment]
that `U _ZTIN14reactphysics3d14CollisionShapeE` is explicitly an undefined symbol
so it is very much not in the static library

```
$ c++filt
_ZTIN14reactphysics3d14CollisionShapeE
typeinfo for reactphysics3d::CollisionShape
```

https://stackoverflow.com/a/307427 -> "One possible reason is because you are declaring a virtual function without defining it."

I would double check that every single virtual function is either defined somewhere (header or some .cpp) or pure virtual `= 0;`
Further, double check the destructor. it's common for virtual types to have `~SomeClass() {}` - perhaps the implementation was empty and removed from the cpp file and ended up with a semicolon in the header without implementation.
